acceptance/steps/helpers.go

```diff
@@ -354,3 +363,116 @@ func checkStableResource(ctx context.Context, t framework.TestingT, o runtimecli
 }
+
+type operatorClients struct {
+	client             http.Client
+	operatorPodName    string
+	namespace          string
+	schema             string
+	token              string
+	expectedStatusCode int
+}
+
+func (c *operatorClients) ExpectRequestRejected(ctx context.Context) {
+	t := framework.T(ctx)
+
+	url := fmt.Sprintf("%s://%s.%s:8443/metrics", c.schema, c.operatorPodName, c.namespace)
+
+	t.Logf("Request %s to operator", url)
+	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+	require.NoError(t, err)
+
+	req.Header.Set("Authorization", "Bearer "+c.token)
+	resp, err := c.client.Do(req)
+	require.NoError(t, err)
+
+	require.Equal(t, c.expectedStatusCode, resp.StatusCode)
+
+	defer resp.Body.Close()
+}
+
+func (c *operatorClients) ExpectCorrectMetricsResponse(ctx context.Context) {
+	t := framework.T(ctx)
+
+	url := fmt.Sprintf("%s://%s.%s:8443/metrics", c.schema, c.operatorPodName, c.namespace)
+
+	t.Logf("Request %s to operator", url)
+	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+	require.NoError(t, err)
+
+	req.Header.Set("Authorization", "Bearer "+c.token)
+	resp, err := c.client.Do(req)
+	require.NoError(t, err)
+
+	require.Equal(t, http.StatusOK, resp.StatusCode)
+
+	defer resp.Body.Close()
+
+	var parser expfmt.TextParser
+	_, err = parser.TextToMetricFamilies(resp.Body)
+	require.NoError(t, err)
+}
+
+func clientsForOperator(ctx context.Context, includeTLS bool, serviceAccountName, expectedStatusCode string) *operatorClients {
+	t := framework.T(ctx)
+
+	var dep appsv1.Deployment
+	require.NoError(t, t.Get(ctx, t.ResourceKey("redpanda-operator"), &dep))
+
+	var podList corev1.PodList
+
+	require.NoError(t, t.List(ctx, &podList, &runtimeclient.ListOptions{
+		LabelSelector: labels.SelectorFromSet(dep.Spec.Selector.MatchLabels),
+	}))
+
+	require.Len(t, podList.Items, 1, "expected 1 pod, got %d", len(podList.Items))
+
+	var tlsCfg tls.Config
+	schema := "http"
+	if includeTLS {
+		tlsCfg = tls.Config{InsecureSkipVerify: includeTLS} // nolint:gosec
+		schema = "https"
+	}
+
+	token := randomString(20)
+	if serviceAccountName != "" {
+		cs, err := kubernetes.NewForConfig(t.RestConfig())
+		require.NoError(t, err)
+		tokenResponse, err := cs.CoreV1().ServiceAccounts(t.Namespace()).CreateToken(ctx, serviceAccountName, &authenticationv1.TokenRequest{}, metav1.CreateOptions{})
+		require.NoError(t, err)
+		token = tokenResponse.Status.Token
+	}
+
+	statusCode := http.StatusOK
+	if expectedStatusCode != "" {
+		var err error
+		statusCode, err = strconv.Atoi(expectedStatusCode)
+		require.NoError(t, err)
+	}
+
+	return &operatorClients{
+		expectedStatusCode: statusCode,
+		token:              token,
+		schema:             schema,
+		namespace:          t.Namespace(),
+		operatorPodName:    podList.Items[0].Name,
+		client: http.Client{Transport: &http.Transport{
+			TLSClientConfig: &tlsCfg,
+			DialContext:     kube.NewPodDialer(t.RestConfig()).DialContext,
+		}},
+	}
+}
+
+func init() {
```

Comment:
This doesn't belong here. If we want to seed the global RNG that should go into `main` or `TestMain`. Mutating globals in `init` can cause a looooooooot of issues.

Reply:
I just want randomString function, so I copied it from https://github.com/redpanda-data/redpanda-operator/blob/054413d40605338549a4fd85786a817599d7fa19/harpoon/internal/testing/strings.go

I will remove that randomString and replace https://github.com/redpanda-data/redpanda-operator/pull/395/files#diff-a3851ab23f5f53bf1cbd57cdccd8c293942bb15708815c64f34afff006316270R437 with 
```go
	token := "non-existen-token"
```

Reply:
This init call still needs to be moved or removed.